workspaces/arborist/lib/arborist/reify.js

```diff
@@ -716,8 +716,16 @@ module.exports = cls => class Reifier extends cls {
     // ${REGISTRY} or something.  This has to be threaded through the
     // Shrinkwrap and Node classes carefully, so for now, just treat
     // the default reg as the magical animal that it has been.
-    return resolved && resolved
-      .replace(/^https?:\/\/registry\.npmjs\.org\//, this.registry)
+    const resolvedURL = new URL(resolved)
+    if (resolved
+      && ((this.options.replaceRegistryHost === 'npmjs'
+        && resolvedURL.hostname === 'registry.npmjs.org')
```

Comment:
well, i mean.. you can break pretty much anything if you DNS-shadow a domain and change the port and other parts of the URL.. that's a very edge case thing to do and one i'm not convinced we need to address, at least not immediately, seeing as that would already be broken today

Reply:
fair enough, just seemed like a good opportunity to avoid hardcoding things.

maybe instead of "npmjs" being the default, the full hostname could be the default?

Reply:
that's an interesting idea. @fritzy what do you think?

Reply:
So... always, never, and <hostname>?

Reply:
and `registry:registry.npmjs.org`?

Reply:
I'll have to make a breaking change on pacote for a feature I just added, but I think you're right. This is worth it.